store/streaming/plugins/abci/v1/grpc.go

```diff
@@ -0,0 +1,148 @@
+			cleanupAndExit()
+		}
+		return err
+	}
+
+	return nil
+}
+
+func (m GRPCClient) ListenDeliverTx(goCtx context.Context, req abci.RequestDeliverTx, res abci.ResponseDeliverTx) error {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	sm := ctx.StreamingManager()
+	request := &ListenDeliverTxRequest{BlockHeight: ctx.BlockHeight(), Req: &req, Res: &res}
+
+	if sm.Async {
+		go func(ctx sdk.Context) {
+			_, err := m.client.ListenDeliverTx(ctx, request)
+			ctx.Logger().Error("DeliverTx listening hook failed", "height", ctx.BlockHeight(), "err", err)
+		}(ctx)
+	} else {
+		_, err := m.client.ListenDeliverTx(ctx, request)
+		if err != nil && sm.StopNodeOnErr {
+			ctx.Logger().Error("DeliverTx listening hook failed", "height", ctx.BlockHeight(), "err", err)
+			cleanupAndExit()
+		}
+		return err
+	}
+
+	return nil
+}
+
+func (m GRPCClient) ListenCommit(goCtx context.Context, res abci.ResponseCommit, changeSet []*storetypes.StoreKVPair) error {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	sm := ctx.StreamingManager()
+	request := &ListenCommitRequest{BlockHeight: ctx.BlockHeight(), Res: &res, ChangeSet: changeSet}
+
+	if sm.Async {
+		go func(ctx sdk.Context) {
+			_, err := m.client.ListenCommit(ctx, request)
+			ctx.Logger().Error("Commit listening hook failed", "height", ctx.BlockHeight(), "err", err)
+		}(ctx)
+	} else {
+		_, err := m.client.ListenCommit(ctx, request)
+		if err != nil && sm.StopNodeOnErr {
+			ctx.Logger().Error("Commit listening hook failed", "height", ctx.BlockHeight(), "err", err)
+			cleanupAndExit()
+		}
+		return err
+	}
+
+	return nil
+}
+
+func cleanupAndExit() {
+	plugin.CleanupClients()
+	os.Exit(1)
+}
+
+var _ ABCIListenerServiceServer = (*GRPCServer)(nil)
+
+// GRPCServer is the gRPC server that GRPCClient talks to.
+type GRPCServer struct {
+	// This is the real implementation
+	Impl storetypes.ABCIListener
+}
+
+func (m GRPCServer) ListenBeginBlock(ctx context.Context, request *ListenBeginBlockRequest) (*ListenBeginBlockResponse, error) {
+	if err := m.Impl.ListenBeginBlock(ctx, *request.Req, *request.Res); err != nil {
+		return nil, err
+	}
+	return &ListenBeginBlockResponse{}, nil
+}
+
+func (m GRPCServer) ListenEndBlock(ctx context.Context, request *ListenEndBlockRequest) (*ListenEndBlockResponse, error) {
+	if err := m.Impl.ListenEndBlock(ctx, *request.Req, *request.Res); err != nil {
+		return nil, err
+	}
+	return &ListenEndBlockResponse{}, nil
+}
+
+func (m GRPCServer) ListenDeliverTx(ctx context.Context, request *ListenDeliverTxRequest) (*ListenDeliverTxResponse, error) {
+	if err := m.Impl.ListenDeliverTx(ctx, *request.Req, *request.Res); err != nil {
+		return nil, err
+	}
+	return &ListenDeliverTxResponse{}, nil
+}
+
+func (m GRPCServer) ListenCommit(ctx context.Context, request *ListenCommitRequest) (*ListenCommitResponse, error) {
+	if err := m.Impl.ListenCommit(ctx, *request.Res, request.ChangeSet); err != nil {
+		return nil, err
+	}
+	return &ListenCommitResponse{}, nil
+}
```

Comment:
All of these methods return empty responses on success, is that intentional?

Reply:
Yes, I don't see a reason to report additional information upstream.

Reply:
Is this acceptable? Can I resolve this conversation?